AirCasting/MainTabBarView.swift

```diff
@@ -54,14 +55,25 @@ struct MainTabBarView: View {
             
         }
         .onAppCameToForeground {
-            measurementUpdatingService.updateAllSessionsMeasurements()
+            measurementsDownloadingInProgress = true
+            measurementUpdatingService.updateAllSessionsMeasurements() {
+                DispatchQueue.main.async {
+                    measurementsDownloadingInProgress = false
+                }
```

Comment:
I think we don't need  `DispatchQueue.main.async` as long as this code is View related

Reply:
Could you elaborate? Cause I thought we need it specifically because it is view related 🤔